.github/workflows/e2e.yaml

```diff
@@ -24,16 +24,10 @@ jobs:
           tenant-id: ${{ secrets.AZURE_TENANT_ID }}
           subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
 
-      - name: "Provision infrastructure"
+      - name: "Run E2E"
         env:
           ARM_USE_OIDC: "true"
           ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
           ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_E2E_SUBSCRIPTION_ID }}
           ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
-        run: "make dev-public"
-
-      - name: "Deploy operator"
-        run: "make push"
-
-      - name: "Run tests"
-        run: "make e2e"
+        run: "make all"
```

Comment:
we should be using a matrix and testing all scenarios

Reply:
that makes more sense than having a bash script that runs the matrix. we can run some things in parallel